django/db/migrations/state.py

```diff
@@ -98,16 +98,32 @@ def __init__(self, models=None, real_apps=None):
         self.real_apps = real_apps
         self.is_delayed = False
         # {remote_model_key: {model_key: [(field_name, field)]}}
-        self.relations = None
+        self._relations = None
+
+    @property
+    def relations(self):
+        if self._relations is None:
+            self.resolve_fields_and_relations()
+        return self._relations
 
     def add_model(self, model_state):
-        app_label, model_name = model_state.app_label, model_state.name_lower
-        self.models[(app_label, model_name)] = model_state
+        model_key = model_state.app_label, model_state.name_lower
+        self.models[model_key] = model_state
+        if self._relations is not None:
+            concretes, _ = self._get_concrete_models_mapping_and_proxy_models()
+            self.resolve_model_relations(model_state, model_key, concretes, )
```

Comment:
You need to add `model_state` to the `resolve_model_relations()` signature. Also, please remove unnecessary commas and spaces:
```suggestion
            self.resolve_model_relations(model_state, model_key)
```
I'm not sure why tests work.

Reply:
I don't think that there is any need to add `model_state` to the `resolve_model_relations()` signature. As we are passing `model_key` which can be used to get the state. I am removing the `model_state` parameter.